packages/driver/src/dom/jquery.js

```diff
@@ -1,6 +1,16 @@
 const $ = require('jquery')
 const _ = require('lodash')
 
+// Check if element contains a child element
+const contains = (doc, obj) => {
+  if (!doc || !obj) {
+    return false
+  }
+
+  // offsetParent works with shadowDom (parent is in dom)
+  return $.contains(doc, obj) || $.contains(doc, obj.offsetParent)
```

Comment:
again offsetParent doesn't really exist for this.

you should check if `getRootNode` is a shadow root and if its `host` contains the target element. repeating this logic upwards until you reach the containing element's root or `document`.

Reply:
here's one i wrote a year or two ago for quilljs:

```ts
    const containsNode = (node, child) => {
      if (child.getRootNode() === document) {
        return node.contains(child);
      }

      while (!node.contains(child)) {
        const root = child.getRootNode();
        if (!root || !root.host) {
          return false;
        }
        child = root.host;
      }

      return true;
    };
```